Проанализируйте результаты эксперимента и напишите свои рекомендации менеджеру.

Mobile Games AB Testing with Cookie Cats

In [1]:
from scipy import stats
import numpy as np

import warnings
warnings.filterwarnings('ignore')
warnings.warn('DelftStack')
warnings.warn('Do not show this message')

import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd

In [51]:
data = pd.read_csv('результаты А_B - cookie_cats 2.csv')

In [52]:
data

userid  version  sum_gamerounds  retention_1  retention_7
0          116  gate_30               3            0            0
1          337  gate_30              38            1            0
2          377  gate_40             165            1            0
3          483  gate_40               1            0            0
4          488  gate_40             179            1            1
...        ...      ...             ...          ...          ...
90184  9999441  gate_40              97            1            0
90185  9999479  gate_40              30            0            0
90186  9999710  gate_30              28            1            0
90187  9999768  gate_40              51            1            0
90188  9999861  gate_40              16            0            0

[90189 rows x 5 columns]

In [53]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90189 entries, 0 to 90188
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   userid          90189 non-null  int64 
 1   version         90189 non-null  object
 2   sum_gamerounds  90189 non-null  int64 
 3   retention_1     90189 non-null  int64 
 4   retention_7     90189 non-null  int64 
dtypes: int64(4), object(1)
memory usage: 3.4+ MB


In [54]:
data.describe()

userid  sum_gamerounds   retention_1   retention_7
count  9.018900e+04    90189.000000  90189.000000  90189.000000
mean   4.998412e+06       51.872457      0.445210      0.186065
std    2.883286e+06      195.050858      0.496992      0.389161
min    1.160000e+02        0.000000      0.000000      0.000000
25%    2.512230e+06        5.000000      0.000000      0.000000
50%    4.995815e+06       16.000000      0.000000      0.000000
75%    7.496452e+06       51.000000      1.000000      0.000000
max    9.999861e+06    49854.000000      1.000000      1.000000

In [55]:
data['version'].value_counts()

gate_40    45489
gate_30    44700
Name: version, dtype: int64

Посмотрим на % юзеров, которые вернулись на следующий день после установки

In [57]:
data['retention_1'].sum() / data['retention_1'].count()

0.4452095044850259

Посмотрим на % юзеров, которые вернулись на следующий день после установки, только уже для каждой из групп

In [58]:
data.groupby('version')['retention_1'].sum() / data.groupby('version')['userid'].count()


version
gate_30    0.448188
gate_40    0.442283
dtype: float64

Мы видим, что между тестовой группой gate_40(44,2%) и контрольной gate_30 (44,8%) разница очень мала. Но в метрике retention все равно это очень важно. Вывод на основании 1 дня сделать невозможно. Посмотрим retention 7 дней.

In [59]:
data.groupby('version')['retention_7'].sum() / data.groupby('version')['userid'].count()

version
gate_30    0.190201
gate_40    0.182000
dtype: float64

Как и в метрике retention 1 дня, разница между тестовой группой gate_40(18,2%) и контрольной gate_30 (19%) небольшая, но она больше, чем при retention 1 дня. Также мы видим, что общий % возвращаемости на 7-й день ниже, чем на 1-й. => Через неделю после установки в игру играет меньшее количество людей, чем через день после установки.

### Вывод: 
% удержания юзеров выше, когда ворота находятся на 30 уровне, чем, когда они находятся на 40 уровне. Тк retention - одна из наиболее важных метрик, все-таки не нужно повышать уровень ворот для пользователей, чтобы сохранить и преумножить нашу базу игроков.